src/gui/base/Overlay.ts

```diff
@@ -95,20 +104,32 @@ export const overlay: Component = {
 							height: position.height,
 							"z-index": position.zIndex != null ? position.zIndex : LayerType.Overlay,
 						},
-						oncreate: (vnode) => {
-							const dom = vnode.dom as HTMLElement
-							overlayAttrs[1] = dom
-
-							if (attrs.createAnimation) {
-								animations.add(dom, attrs.createAnimation(dom))
+						onanimationend: () => {
+							if (attrs.isClosing) {
+								overlays.delete(key)
 							}
 						},
-						onremove: () => {
-							overlayAttrs[1] = null
+						onupdate(vnode: VnodeDOM<any>): any {
+							if (attrs.isClosing && attrs.closeAnimation != null) {
+								const dom = vnode.dom as HTMLElement
+
+								// Force the environment to restart the animations via a reflow
+								dom.className = baseClasses
+								void dom.offsetWidth
+								dom.className = classes
+							}
 						},
 					},
 					m(attrs.component),
 				)
 			}),
 		),
 }
+
+function mapToArray<T, K>(map: Map<T, K>): Array<[T, K]> {
```

Comment:
Just wanted to note here that I wanted to put this into the map utils package but was not sure how.

Reply:
so is it like `Array.from(map.entries())`?

Reply:
TypeScript gave me warnings about it not being compatible with ECMAScript 2015 when I tried that.

Reply:
Ah, my mistake I tried `overlay.entries().map()` which is not supported yet on `IterableIterator`s. I just peeked at the `tsconfigs`s & it seems we are targeting ECMAScript 2020. So I have no idea why it showed an error about ECMAScript 2015 specifically.

I will remove this function now.